session/sessionstate.go

```diff
@@ -562,7 +567,16 @@ func (state *State) TriggerEvents(actionState *action.State, chHandles, clHandle
 
 // SendRequest and trigger any object changes synchronously
 func (state *State) SendRequest(actionState *action.State, f func(ctx context.Context) error) error {
-	ctx, cl, cancel := state.ReqContext()
+	return state.SendRequestWithTimeout(actionState, 0, f)
+}
+
+// SendRequestWithTimeout is the same as SendRequest but with a custom timeout
+func (state *State) SendRequestWithTimeout(actionState *action.State, timeout time.Duration, f func(ctx context.Context) error) error {
+	if timeout < 1 {
```

Comment:
Why not check for "0" instead of < 1?
Due time.Duration implementation?

Reply:
time.Duration is a int64 so checking for ==0 is possible. but int64 can technically also be negative so good coding practice to check anything 0 and below. On a general note also note also leads to more secure code.